### PR TITLE
PUB-2162 Add authorisation check to update and delete account endpoints

### PR DESCRIPTION
**Pull request by KianKwa**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/PUB-2162

### Change description ###

Add authorisation check to update and delete account endpoints

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/pip/account/management/controllers/AccountController.java [hunk 1]
```diff
@@ -133,16 +134,32 @@ public ResponseEntity<String> deleteAccount(@PathVariable UUID userId) {
...
     @ApiResponse(responseCode = NOT_FOUND_ERROR_CODE, description = "User not found")
-    @ApiResponse(responseCode = FORBIDDEN_ERROR_CODE, description = "User with id %s is unable to update user ID %s")
+    @ApiResponse(responseCode = FORBIDDEN_ERROR_CODE,
+        description = "User with ID %s is forbidden to update user with ID %s")
     @Operation(summary = "Update a users role by their id")
     @PutMapping("/update/{userId}/{role}")
+    @PreAuthorize("@authorisationService.userCanUpdateAccount(#userId, #adminUserId)")
     public ResponseEntity<String> updateAccountRoleById(@PathVariable UUID userId,
                                                         @PathVariable Roles role,
                                                         @RequestHeader(value = "x-admin-id", required = false)
```
Comment (ChrisS1512) [inline src/main/java/uk/gov/hmcts/reform/pip/account/management/controllers/AccountController.java:161]: Can required now be true for this?

File: src/main/java/uk/gov/hmcts/reform/pip/account/management/service/AuthorisationService.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+                String.format("User with ID %s is forbidden to remove user with ID %s", adminUserId, userId)
+            ));
+        }
+        return isAuthorised;
+    }
+
+    public boolean userCanUpdateAccount(UUID userId, UUID adminUserId) {
+        if (adminUserId != null && adminUserId.equals(userId)) {
+            log.error(writeLog(
+                String.format("User with ID %s is forbidden to update their owned account", userId)
```
Comment (ChrisS1512) [inline src/main/java/uk/gov/hmcts/reform/pip/account/management/service/AuthorisationService.java:37]: I think this should be 'own' rather than 'owned'

**Review by ChrisS1512 — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/pip/account/management/controllers/AccountController.java (hunk 1 above)
Comment (KianKwa) [reply src/main/java/uk/gov/hmcts/reform/pip/account/management/controllers/AccountController.java]: Removed required=false to use the default true value

File: src/main/java/uk/gov/hmcts/reform/pip/account/management/service/AuthorisationService.java (hunk 2 above)
Comment (KianKwa) [reply src/main/java/uk/gov/hmcts/reform/pip/account/management/service/AuthorisationService.java]: Corrected typo

**Review by ChrisS1512 — APPROVED** [review]

**Pull request merged**